plugins/scaffolder-backend/src/service/router.ts

```diff
@@ -84,6 +85,12 @@ export async function createRouter(
       const values: RequiredTemplateValues & Record<string, JsonValue> =
         req.body.values;
 
+      const validationResult: ValidatorResult = validate(values, template.spec.schema);
+      if (!validationResult.valid) {
+        res.status(400).json({ errors: validationResult.errors });
```

Comment:
We could just return validationResult.toString() which creates a human readable list but I've chosen this as I think something more machine readable makes sense here given it's most likely this will be hit during API usage.